src/components/DocSearch/DocSearch.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * @emails react-core
+ */
+
+import React, {Component} from 'react';
+import {colors, media} from 'theme';
+
+class DocSearch extends Component {
+  state = {
+    disabled: false,
+  };
+  componentDidMount() {
+    // Initialize Algolia search.
+    // TODO Is this expensive? Should it be deferred until a user is about to search?
+    // eslint-disable-next-line no-undef
+    if (window.docsearch) {
+      window.docsearch({
+        apiKey: '36221914cce388c46d0420343e0bb32e',
+        indexName: 'react',
+        inputSelector: '#algolia-doc-search',
+      });
+    } else {
+      this.setState({disabled: true});
```

Comment:
Might be nice to add a `console.warn` that search failed to load and so it's being disabled.